web/html/src/core/spa/spa-engine.tsx

```diff
@@ -73,14 +76,22 @@ window.pageRenderers.spaengine.init = function init() {
       jQuery(".modal-backdrop.removeWhenNavigationEnds").remove();
 
       // If an error happens we make a full refresh to make sure the original request is shown instead of a SPA replacement
-      if (
-        navigation.error &&
-        (navigation.error.statusCode === 401 ||
-          navigation.error.invalidStatus ||
-          navigation.error.timeout ||
-          navigation.error.requestError)
-      ) {
-        window.location = navigation.path;
+      if (navigation.error) {
+        if (navigation.error.statusCode === 401 ||
+             navigation.error.invalidStatus ||
+             navigation.error.requestError) {
+          window.location = navigation.path;
+        } else if (navigation.error.timeout) {
+          // Stop loading bar
+          jQuery(document.documentElement).removeClass('senna-loading')
+          // Inform user that page must be reloaded
+          const message = (
+            <>
+                Request has timed out, please <a className="btn-link" onClick={() => window.location = navigation.path}>reload the page</a>.
```

Comment:
If this is a link anyway, it should be possible to use `href={navigation.path}` instead of `onClick={...}`.  

This is something that we have in older code here and there, but would be nice to avoid in new code. In short, things that navigate should be links and use `href`, and things that trigger actions should be buttons and use `onClick` and similar.

Reply:
Note that `href` only works when navigating to an URL that is different from the current page. So I've changed it to a button.